addons/viewport/src/components/tests/Panel.spec.jsx

```diff
@@ -0,0 +1,249 @@
+/* global document */
+import React from 'react';
+import { shallow } from 'enzyme'; // eslint-disable-line import/no-extraneous-dependencies
+
+import { Panel } from '../Panel';
+import { viewports, defaultViewport, resetViewport } from '../viewportInfo';
+import * as styles from '../styles';
+
+describe('Viewport/Panel', () => {
+  const props = {
+    channel: {
+      on: jest.fn(),
+      removeListener: jest.fn(),
+    },
+  };
+
+  let subject;
+
+  beforeEach(() => {
+    subject = shallow(<Panel {...props} />);
+  });
+
+  describe('construct', () => {
+    it('creates the initial state', () => {
+      expect(subject.instance().state).toEqual({
+        viewport: defaultViewport,
+        isLandscape: false,
+      });
+    });
+
+    it('listens on the channel', () => {
+      expect(props.channel.on).toHaveBeenCalledWith(
+        'addon:viewport:update',
+        subject.instance().changeViewport
+      );
+    });
+  });
+
+  describe('componentDidMount', () => {
+    let previousGet;
+
+    beforeEach(() => {
```

Comment:
low priority:

I prefer to use `beforeEach` and `afterEach` functions as last resort only.
The reason being it becomes harder and hard to know/understand all the steps executed to setup a test.

Not directly necessary to get this PR merged, but I'd like to see a `setup` function being used to get a shallow render in a specific state.